autoblocks/_impl/configs/config.py

```diff
@@ -0,0 +1,246 @@
+
+    raise ValueError(f"Invalid config type: {config}")
+
+
+def is_remote_config_refreshable(config: RemoteConfig) -> bool:
+    """
+    A config is refreshable if the user has specified that they want the latest deployed config
+    or the latest undeployed config.
+    """
+    return isinstance(config, LatestRemoteConfig) or (
+        isinstance(config, DangerouslyUseUndeployedRemoteConfig)
+        and isinstance(config.dangerouslyUseUndeployed, LatestDangerouslyUseUndeployed)
+    )
+
+
+async def get_remote_config(config: RemoteConfig, timeout: timedelta, api_key: str) -> RemoteConfigResponse:
+    resp = await global_state.http_client().get(
+        make_request_url(config=config),
+        timeout=timeout.total_seconds(),
+        headers={"Authorization": f"Bearer {api_key}"},
+    )
+    resp.raise_for_status()
+    return RemoteConfigResponse(
+        id=resp.json().get("id"),
+        version=resp.json().get("version"),
+        value=resp.json().get("value"),
+    )
+
+
+class AutoblocksConfig(
+    abc.ABC,
+    Generic[AutoblocksConfigValue],
+):
+    _value: AutoblocksConfigValue
+
+    def __init__(self, value: AutoblocksConfigValue) -> None:
+        self._value = value
+
+    async def _refresh_loop(
+        self,
+        config: RemoteConfig,
+        refresh_interval: timedelta,
+        refresh_timeout: timedelta,
+        api_key: str,
+        parser: Callable[[Dict[str, Any]], AutoblocksConfigValue],
+    ) -> None:
+        """
+        Refreshes the latest version of the config every `refresh_interval` seconds.
+
+        This isn't exactly "every n seconds" but more "run and then wait n seconds",
+        which will cause it to drift each run by the time it takes to run the refresh,
+        but I think that is ok in this scenario.
+        """
+        while True:
+            await asyncio.sleep(refresh_interval.total_seconds())
+
+            try:
+                await self._load_and_set_remote_config(
+                    config=config, api_key=api_key, timeout=refresh_timeout, parser=parser
+                )
+            except Exception as err:
+                log.warning(f"Failed to refresh config '{config.id}': {err}")
+
+    async def _load_and_set_remote_config(
+        self,
+        config: RemoteConfig,
+        api_key: str,
+        timeout: timedelta,
+        parser: Callable[[Dict[str, Any]], AutoblocksConfigValue],
+    ) -> None:
+        """
+        Loads the remote config from Autoblocks and sets the value of this config
+        based on the result.
+
+        If the parser is specified, the value will be parsed using the parser.
+        """
+        remote_config = await get_remote_config(config, timeout=timeout, api_key=api_key)
+        try:
+            parsed = parser(remote_config.value)
+            if parsed is not None:
+                self._value = parsed
+        except Exception as e:
+            raise Exception(f"Failed to parse config '{config.id}': {e}")
```

Comment:
i would log an error but re-raise the original instead of turning it into a new exception:

```python
except Exception as e:
    log.error(f"Failed to parse config '{config.id}': {e}", exc_info=True)
    raise
```